modules/yugabytedb/yugabytedb_test.go

```diff
@@ -0,0 +1,108 @@
+package yugabytedb_test
+
+import (
+	"context"
+	"database/sql"
+	"fmt"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"github.com/yugabyte/gocql"
+
+	_ "github.com/lib/pq"
+	"github.com/testcontainers/testcontainers-go"
+	"github.com/testcontainers/testcontainers-go/modules/yugabytedb"
+)
+
+func TestYugabyteDB_YSQL(t *testing.T) {
+	t.Run("Run", func(t *testing.T) {
+		ctx := context.Background()
+
+		ctr, err := yugabytedb.Run(ctx, "yugabytedb/yugabyte:2024.1.3.0-b105")
+		testcontainers.CleanupContainer(t, ctr)
+		require.NoError(t, err)
+
+		ctrHost, err := ctr.Host(ctx)
+		require.NoError(t, err)
+
+		ctrPort, err := ctr.MappedPort(ctx, "5433/tcp")
+		require.NoError(t, err)
+
+		ysqlConnStr, err := ctr.YSQLConnectionString(ctx, "sslmode=disable")
+		require.NoError(t, err)
+		assert.Equal(t, fmt.Sprintf("postgres://yugabyte:yugabyte@%s:%s/yugabyte?sslmode=disable", ctrHost, ctrPort.Port()), ysqlConnStr)
+
+		db, err := sql.Open("postgres", ysqlConnStr)
+		require.NoError(t, err)
+		require.NotNil(t, db)
+
+		err = db.Ping()
+		require.NoError(t, err)
+	})
+
+	t.Run("custom-options", func(t *testing.T) {
```

Comment:
bug: when I run this test, it always fail, I guess because, when a database is set, the wait strategy should include a condition for the database is ready to accept connections. I _think_ this is the case, because if I debug the test, step by step, it always pass, so it gives me the sense it's a race condition.

```
    /Users/mdelapenya/sourcecode/src/github.com/testcontainers/testcontainers-go/modules/yugabytedb/yugabytedb_test.go:69:
        	Error Trace:	/Users/mdelapenya/sourcecode/src/github.com/testcontainers/testcontainers-go/modules/yugabytedb/yugabytedb_test.go:69
        	Error:      	Received unexpected error:
        	            	pq: password authentication failed for user "custom-user"
        	Test:       	TestYugabyteDB_YSQL/custom-options
```

Reply:
Good catch. Only ran it in debug mode which never failed. I've added an additional log constraint which works on my side.